onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -162,6 +162,47 @@ void convTransposeWithDynamicPadsShapeInference(InferenceContext& ctx) {
     return;
   }
 }
+
+void embedLayerNormalizationShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 2, 0);
+  propagateElemTypeFromInputToOutput(ctx, 0, 1);
+  if (!hasInputShape(ctx, 0))
+    return;
```

Comment:
Nit: Could still return a 3D shape with the `hidden_size` in this case.

Reply:
Is there a way to update the input shape? I only see `updateOutputShape()` in `shape_inference.h`:

https://github.com/onnx/onnx/blob/adeb09b3af63939c507355f1ef6bca9c32e7e244/onnx/defs/shape_inference.h

Reply:
I don't think there is, but I was suggesting something slightly different... if the input doesn't have a shape, we can still call `updateOutputShape()` with the shape `(?, ?, hidden_size)`.

(Not really the topic of this PR, but there have been quite a few cases in ONNX where the shape inference function bails out earlier than it strictly needs to, and then we lose out on the opportunity to do a shape-based optimization further downstream - many of these make progress even without complete info.)